invenio_rdm_records/services/communities/service.py

```diff
@@ -37,10 +54,109 @@ def record_cls(self):
+        if existing_request_id:
+            raise OpenRequestAlreadyExists(existing_request_id)
+
+        type_ = current_request_type_registry.lookup(CommunityInclusion.type_id)
+        receiver = ResolverRegistry.resolve_entity_proxy(
+            {"community": community_id}
+        ).resolve()
+
+        empty_data = {}
+        request_item = current_requests_service.create(
+            identity,
+            empty_data,
+            type_,
+            receiver,
+            topic=record,
+            uow=uow,
+        )
+        # create review request
+        request_item = current_rdm_records.community_inclusion_service.submit(
+            identity, record, community, request_item._request, empty_data, uow
+        )
+        # include directly when allowed
+        return current_rdm_records.community_inclusion_service.include(
+            identity, community, request_item._request, uow
+        )
+
     @unit_of_work()
-    def create(self, identity, data, record, uow=None):
+    def add(self, identity, id_, data, uow):
         """Include the record in the given communities."""
-        pass
+        valid_data, errors = self.schema.load(
+            data,
+            context={
+                "identity": identity,
+                "max_number": self.config.max_number_of_additions,
+            },
+            raise_errors=True,
+        )
+        communities = valid_data["communities"]
+
+        record = self.record_cls.pid.resolve(id_)
+        self.require_permission(identity, "add_community", record=record)
+
+        success = []
+        for community in communities:
+            community_id = community["id"]
+
+            result = {
+                "community": community_id,
+            }
+            try:
+                request_item = self._include(identity, community_id, record, uow)
+                result["request"] = str(request_item.data["id"])
+                success.append(result)
+            except (NoResultFound, PIDDoesNotExistError):
+                result["message"] = _("Community not found.")
+                errors.append(result)
+            except CommunityAlreadyExists:
+                result["message"] = _(
+                    "The record is already included in this community."
+                )
+                errors.append(result)
+            except OpenRequestAlreadyExists:
+                result["message"] = _(
+                    "There is already an open inclusion request for this community."
+                )
+                errors.append(result)
+            except CommunityInclusionInconsistentAccessRestrictions as ex:
+                result["message"] = ex.args[0]
+                errors.append(result)
+            except PermissionDeniedError:
+                result["message"] = _("Permission denied.")
+                errors.append(result)
+
+        uow.register(IndexRefreshOp(indexer=self.indexer))
+
+        return dict(success=success, errors=errors)
```

Comment:
do we still discuss on Monday?

Reply:
I think the best would be to discuss it and the @jrcastro2 can make changes in his next PR ;)